### PR TITLE
Updated kubernetes example

### PR DESCRIPTION
**Pull request by nklmilojevic**
- Updated go runtime and updated golang dependencies
- Added correct docker image url
- Update sample.yaml so it works with k8s >1.16
- Fixed some commands in readme

### CODE REVIEW

**Comment by nklmilojevic** [comment]
@googlebot I signed it!

File: examples/kubernetes/main.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 const (
 	// berglasContainer is the default berglas container from which to pull the
 	// berglas binary.
-	berglasContainer = "us-docker.pkg.dev/berglas/berglas/berglas:latest"
+	berglasContainer = "gcr.io/berglas/berglas:latest"
```
Comment (sethvargo) [inline examples/kubernetes/main.go:20]: I'm fine with the other changes, but the gcr.io URLs are behind and will not receive future updates. We're using Artifact Registry for berglas moving forward.

Comment (nklmilojevic) [reply examples/kubernetes/main.go:20]: Alright, but pulling from those images doesn't work:

https://github.com/GoogleCloudPlatform/berglas/issues/128

Comment (nklmilojevic) [reply examples/kubernetes/main.go:20]: Okay. It works now. It didn't work for the whole day yesterday, hence the changes :)

Comment (nklmilojevic) [reply examples/kubernetes/main.go]: Restored it back to the correct repo.

**Pull request merged**